doc/storagedriver/azure.md

```diff
@@ -10,6 +10,7 @@ The following parameters must be used to authenticate and configure the storage
 * `accountname`: Name of the Azure Storage Account.
 * `accountkey`: Primary or Secondary Key for the Storage Account.
 * `container`: Name of the root storage container in which all registry data will be stored. Must comply the storage container name [requirements][create-container-api].
+* `domain`: (optional) Domain name for the Storage Service. Defaults to `core.windows.net`. For example, `core.chinacloudapi.cn` would be the correct value for Azure in China and `core.usgovcloudapi.net` would be the correct value for Azure Government.
```

Comment:
@stevvooe problem here is we're not referring to a full hostname by this parameter as I described above, that's just a segment of it which happens to be the suffix of the hostname as well. API docs don't call it anything specific, really and I don't think it's anything related to Go or registry, it just something specific to Azure. Thinking more about it, `realm` or `hostsuffix` makes more sense.